Services/communication_service.c

```diff
@@ -35,31 +53,753 @@
 SAT_returnState communication_service_app(csp_packet_t *packet) {
   uint8_t ser_subtype = (uint8_t)packet->data[SUBSERVICE_BYTE];
   int8_t status;
-  struct temp_utc temp_temp;
+  int SID, i;
```

Comment:
if `i` is used as an index variable, it should proably be declared directly above the loop where it's used for clarity.

Reply:
Yeah, my bad.